### PR TITLE
Feature/ Implement pagination component

### PR DESCRIPTION
**Pull request by JennyDoJD**
Result (UPDATED):
image

image

image

In case of 1 page:
image

Ticket: [Implement Pagination component](https://trello.com/c/yQtxWk4M/33-implement-pagination-component)

### CODE REVIEW

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+  };
+
+  const renderButtons = () => {
+    const buttons = [];
+
+    for (let i = 1; i <= totalPages; i++) {
+      buttons.push(
+        <Button
+          key={i}
```
Comment (ghost) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx]: @JennyDoJD Don't use index as key 
Refer https://dev.to/shiv1998/why-not-to-use-index-as-key-in-react-lists-practical-example-3e66.
You should do more research on this

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  const handleClick = (page) => {
+    if (page !== currentPage) {
+      onPageChange(page);
+    }
+  };
+
+  const renderButtons = () => {
+    const buttons = [];
+
+    for (let i = 1; i <= totalPages; i++) {
```
Comment (ghost) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx]: @JennyDoJD how about if we have 100 pages?

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx (hunk 1 above)
Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: I fixed it

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    return pages;
+  };
+
+  const renderButtons = () => {
+    const pages = getPageNumbers();
+
+    return pages.map((page) => (
+      <Button
+        key={page}
```
Comment (ghost) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx]: It is the same as the previous code, you should read the article again and ensure you understand why we should avoid using key as index(1, 2, 3...)

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx (hunk 2 above)
Comment (ghost) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: @JennyDoJD did you check my comment?

Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: I got it. I'm fixing your comment.

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx (hunk 3 above)
Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: I done

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx (hunk 2 above)
Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: image

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    if (page !== currentPage && page >= 1 && page <= totalPages) {
+      onPageChange(page);
+    }
+  };
+
+  const getPageNumbers = () => {
+    const pages = [];
+    const ellipsis = '...';
+
+    if (totalPages <= 5) {
```
Comment (ghost) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx:28]: @JennyDoJD can we reduce the if else statement?

Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx:28]: Maybe it's okay. I will refactor it.

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+      {currentPage > 1 && (
+        <Button
+          label="Previous"
+          variant="primary"
+          onClick={() => handleClick(currentPage - 1)}
+          disabled={currentPage === 1}
+        />
+      )}
+      {renderButtons()}
+      {currentPage < totalPages && (
```
Comment (ghost) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx]: @JennyDoJD can we move this logic into `renderButtons` function?

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 6]
```diff
@@ -0,0 +1,99 @@
+/* Import dependencies */
+import PropTypes from 'prop-types';
+
+/* Import components */
+import Button from '../Buttons/Button';
+
+/* Import CSS */
+import './pagination.css';
+
+const Pagination = ({ currentPage, totalRecords, pageLimit, onPageChange }) => {
```
Comment (ghost) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx]: @JennyDoJD should we need default value for props?

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx (hunk 5 above)
Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: I done

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx (hunk 6 above)
Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: I done

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx (hunk 4 above)
Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx:28]: I refactored it

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    const ellipsis = '...';
+
+    if (totalPages <= 5) {
+      return Array.from({ length: totalPages }, (_, i) => i + 1);
+    }
+
+    const leftBound = Math.max(1, currentPage - 1);
+    const rightBound = Math.min(totalPages, currentPage + 1);
+
+    if (currentPage <= 3) {
```
Comment (canhpham-agilityio) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx:35]: @JennyDoJD should

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+      pages.push(1, 2, 3, 4, ellipsis, totalPages);
```
Comment (ghost) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx]: Can we return pages at here?

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    } else if (currentPage >= totalPages - 2) {
+      pages.push(
```
Comment (ghost) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx]: similar above comment

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx (hunk 8 above)
Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: Yes

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx (hunk 9 above)
Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: I done

File: practice/react-practice-one/src/components/Pagination/Pagination.jsx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+        totalPages - 3,
+        totalPages - 2,
+        totalPages - 1,
+        totalPages
+      );
+
+      return pages;
+
+      // If none of the above cases apply, calculate the middle pages.
+    } else {
```
Comment (ghost) [inline practice/react-practice-one/src/components/Pagination/Pagination.jsx]: how about this else

Comment (JennyDoJD) [reply practice/react-practice-one/src/components/Pagination/Pagination.jsx]: I fixed it

**Review by ghost — APPROVED** [review]

**Pull request merged**